File: general-client/src/test/java/com/song/middleware/client/GeneralClientTest.java
package com.song.middleware.client;

import com.alibaba.fastjson.JSONObject;
import com.song.middleware.client.util.InetUtils;
import java.util.ArrayList;
import java.util.Collections;
import org.junit.Test;

/**
 * Created by song on 2017/8/6.
 */
public class GeneralClientTest {

    @Test
    public void register() throws Exception {
        GeneralClient generalClient = new GeneralClient();
        generalClient.setServers(new ArrayList<>(Collections.singletonList("localhost:8080")));
        generalClient.setName("general-service");
        generalClient.setIp(InetUtils.findFirstNonLoopbackAddress().getHostAddress());
        generalClient.setPort("8050");
//        generalClient.init();
        System.out.println(JSONObject.toJSON(generalClient.getServices("general-service")));
    }

}

File: general-client/src/test/java/com/song/middleware/client/GeneralClientTest.kt
package com.song.middleware.client

import com.alibaba.fastjson.JSONObject
import com.song.middleware.client.util.InetUtils
import org.junit.Test

/**
 * Created by song on 2017/8/6.
 */
class GeneralClientTest {

    @Test
    @Throws(Exception::class)
    fun register() {
        val generalClient = GeneralClient()
        generalClient.servers = listOf("localhost:8080")
        generalClient.name = "general-service"
        generalClient.ip = InetUtils.findFirstNonLoopbackAddress()?.hostAddress
        generalClient.port = "8050"
        generalClient.init()
        println(JSONObject.toJSON(generalClient.getServices("general-service")))
    }

}